### PR TITLE
Exclude gallery params from caption string

### PR DESCRIPTION
**Pull request by hakubo**
https://wikia-inc.atlassian.net/browse/CONCF-8

This has to be tested across skins.

It was always broken, it's just that Mercury displays this data whereas other skins did not.

### CODE REVIEW

File: extensions/wikia/WikiaPhotoGallery/WikiaPhotoGallery.class.php [hunk 1]
```diff
@@ -118,6 +118,21 @@ function __construct() {
 			'hideoverflow' => false,
 			'sliderbar' => array('bottom','left')
 		);
+
+		$this->mAvalaibleUniqueParams = array_values(
```
Comment (rogatty) [inline extensions/wikia/WikiaPhotoGallery/WikiaPhotoGallery.class.php]: It's misspelled, should be `mAvailableUniqueParams`. And yes, `mAvalaibleGalleryParams` is misspelled too.

Comment (rogatty) [inline extensions/wikia/WikiaPhotoGallery/WikiaPhotoGallery.class.php]: Also, should it be declared like https://github.com/Wikia/app/pull/6300/files#diff-6ddf00ef82821d11d4d1cffd5bd8a74cR56?

File: extensions/wikia/WikiaPhotoGallery/WikiaPhotoGallery.class.php [hunk 2]
```diff
@@ -402,7 +418,12 @@ public function parse(&$parser = null) {
 				} else if (substr($part, 0, 10) == 'shorttext=') {
 					$shorttext = substr($part, 10);
 				} else {
-					$captionParts[] = trim($part);
+					$tempPart = trim($part);
+
+					//If it looks like Gallery param don't treat it as a caption part
+					if (!in_array($tempPart, $this->mAvalaibleUniqueParams)) {
```
Comment (rogatty) [inline extensions/wikia/WikiaPhotoGallery/WikiaPhotoGallery.class.php]: Does it mean that we won't display captions like _small_, _landscape_ etc.? Is it working the same way in MediaWiki?

**Comment by hakubo** [comment]
@rogatty I am not entirely sure, that why it needs some good QA expertise...

**Comment by rogatty** [comment]
LGTM

**Pull request merged**